vumi/components/message_store_cache.py

```diff
@@ -389,26 +385,21 @@ def get_from_addrs(self, batch_id, asc=False):
 
     def count_from_addrs(self, batch_id):
         """
-        Return the number of from_addrs for this batch_id
+        Return count of the unique from_addrs in this batch. Note that the
+        returned count is not exact. We use Redis's HyperLogLog functionality,
+        so the count is subject to a standard error of 0.81%:
+        http://redis.io/commands/pfcount
         """
-        # NOTE: Disabled because this doesn't scale to large batches.
-        #       See https://github.com/praekelt/vumi/issues/877
-
-        # return self.redis.zcard(self.from_addr_key(batch_id))
-        return 0
+        return self.redis.pfcount(self.from_addr_key(batch_id))
 
     def add_to_addr(self, batch_id, to_addr, timestamp):
```

Comment:
No, we can remove it. There shouldn't be anything outside the message store using this.